boards/BMS/Src/App/states/App_PreChargeState.c

```diff
@@ -1,14 +1,40 @@
 #include "states/App_AllStates.h"
+#include "states/App_PreChargeState.h"
+#include "App_AccumulatorVoltages.h"
+#include "Io_VoltageSense.h"
+#include "states/App_DriveState.h"
+#include "Io_PreCharge.h"
+#include "states/App_FaultState.h"
+#include <math.h>
+#include "Io_Airs.h"
+#include "states/App_InitState.h"
 
 #include "App_SetPeriodicCanSignals.h"
 #include "App_SharedMacros.h"
 
+#define S_TO_MS (1000U)
+
+//expected charge time (to 98%) = 4RC * 1000ms/s
+#define PRECHARGE_RESISTANCE (1000U)
+#define PRECHARGE_CAPACITANCE (0.68e-3f)
```

Comment:
Just curious, where did you get PRECHARGE_RESISTANCE AND PRECHARGE_CAPACITANCE from?

Reply:
Precharge resistance is the precharge resistor 1Kohm. Don't think that we had measured the bus cap yet so this value may be a value from the previous car

Reply:
this is the testing cap I have been using. The actual cap is 0.56e-3. I will change it once I verify the code

Reply:
ahhh factsss